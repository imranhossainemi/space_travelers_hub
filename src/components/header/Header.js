import React from 'react';
import { Link } from 'react-router-dom';
import './header.css';
import logo from './planet.png';

const Header = () => (
  <>
    <header className="header">
      <Link className="main-logo" to="/">
        <img className="logo" src={logo} alt="Space Logo" />
        <h3 className="logo-title">Space Travelers&rsquo; Hub</h3>
      </Link>
      <ul className="nav-links">
        <li className="nav-link"><Link to="/">ROCKETS</Link></li>
        <li className="nav-link"><Link to="/missions">MISSIONS</Link></li>
        <li className="nav-link"><Link to="/profile">MY PROFILE</Link></li>
      </ul>
    </header>
    <div className="divider">
      <hr />
    </div>
  </>
);

export default Header;
