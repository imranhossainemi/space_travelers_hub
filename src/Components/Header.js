import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import './header.css';
import logo from '../images/planet.png';

const Header = () => {
  const [activeLink, setActiveLink] = useState('rockets');

  const handleClick = (link) => {
    setActiveLink(link);
  };

  return (
    <>
      <header className="header">
        <Link className="main-logo" to="/">
          <img className="logo" src={logo} alt="Space Logo" />
          <h3 className="logo-title">Space Travelers&rsquo; Hub</h3>
        </Link>
        <ul className="nav-links">
          <li className={`nav-link ${activeLink === 'rockets' ? 'active-nav' : ''}`}>
            <Link to="/" onClick={() => handleClick('rockets')}>
              ROCKETS
            </Link>
          </li>
          <li className={`nav-link ${activeLink === 'missions' ? 'active-nav' : ''}`}>
            <Link to="/missions" onClick={() => handleClick('missions')}>
              MISSIONS
            </Link>
          </li>
          <li className={`nav-link ${activeLink === 'profile' ? 'active-nav' : ''}`}>
            <Link to="/profile" onClick={() => handleClick('profile')}>
              MY PROFILE
            </Link>
          </li>
        </ul>
      </header>
      <div className="divider">
        <hr />
      </div>
    </>
  );
};

export default Header;
