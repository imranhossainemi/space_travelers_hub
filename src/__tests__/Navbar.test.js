import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Header from 'Components/header/Header';

it('test navbar render', () => {
  render(
    <Router>
      <Header />
    </Router>,
  );
  const tree = renderer
    .create(
      <Router>
        <Header />
      </Router>,
    )
    .toJSON();
  const navbars = screen.getAllByRole('link');
  expect(navbars).toHaveLength(3);
  expect(tree).toMatchSnapshot();
});
