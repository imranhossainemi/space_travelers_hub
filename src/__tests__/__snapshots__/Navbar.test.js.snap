// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test navbar render 1`] = `
<nav>
  <p
    className="logo"
  >
    <img
      alt="logo"
      className="logo-image"
      src="planet.png"
    />
    <span
      className="logo-name"
    >
      Space Travelers' Hub
    </span>
  </p>
  <div
    className="navLinks"
  >
    <a
      aria-current="page"
      className="link active"
      href="/"
      onClick={[Function]}
    >
      Rockets
    </a>
    <a
      className="link"
      href="/missions"
      onClick={[Function]}
    >
      My Missions
    </a>
    <a
      className="link"
      href="/myprofile"
      onClick={[Function]}
    >
      My Profile
    </a>
  </div>
</nav>
`;
